crates/red_knot_python_semantic/resources/mdtest/binary/classes.md

```diff
@@ -0,0 +1,27 @@
+# Binary operations on classes
+
+## Union of two classes
+
+Unioning two classes via the `|` operator is only available in Python 3.10 and later.
+
+```toml
+[environment]
+python-version = "3.10"
+```
+
+```py
+class A: ...
+class B: ...
+
+reveal_type(A | B)  # revealed: Literal[A, B]
```

Comment:
I don't think `Literal[A, B]` is the right type to assign to the value expression `A | B`.

In a type expression, `A | B` spells the type `Type::Union[Type::Instance(A), Type::Instance(B)]` -- that is, the type consisting of all instances of `A` (or any subclass) and all instances of `B` (or any subclass).

The value expression `A if flag else B` would correctly be assigned the type `Literal[A, B]`, since its runtime value must either be the class object `A` or the class object `B`.

But the runtime value of the value expression `A | B` is not either of those, so its type can't be `Literal[A, B]`. Its runtime value is an instance of `types.UnionType`, which can't be treated as if it were a class object at all. So the type we should assign to it is `Type::Instance(types.UnionType)`.
```suggestion
reveal_type(A | B)  # revealed: UnionType
```

If we did any special-casing here, the correct special casing would be to create a custom type representation for `types.UnionType` so we can record the fact that its `__args__` attribute (in this case) has the type `tuple[Literal[A], Literal[B]]`. But I don't think this is worth doing.

Reply:
Reverted the special case, this is now returning `UnionType` again as before